matron/src/device/device_crow.c

```diff
@@ -47,10 +49,15 @@ int dev_crow_init(void *self) {
     read(d->fd, s, 255); // clear buffer
     write(d->fd,"^^i\n\0",5);    
     usleep(1000);
-    read(d->fd, s, 255);
-    //fprintf(stderr,"crow init> %i %s",len,s);
-
-    if(strstr(s,"^^identity")==NULL) {
+    int retry = 0;
+    while(retry < CROW_RETRIES) {
```

Comment:
do you want a short wait (`usleep`) in the loop?  or is the `read` blocking enough?